x-pack/plugins/reporting/export_types/common/execute_job/add_force_now_query_string.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+// @ts-ignore
+import url from 'url';
+import { ConditionalHeaders, KbnServer, ReportingJob } from '../../../types';
+import { getAbsoluteUrlFactory } from './get_absolute_url';
+
+function getSavedObjectAbsoluteUrl(job: ReportingJob, relativeUrl: string, server: KbnServer) {
+  const getAbsoluteUrl: any = getAbsoluteUrlFactory(server);
+
```

Comment:
Is it common to handle the "happy path" in an `if` statement and throw at the bottom? I've typically seen error-path's be captured in an `if` block and the "happy path" follows below it. Similar to line 34 below. Example:

```
  if (!relativeUrl) {
    throw new Error(`Unable to generate report. Url is not defined.`);
  }
  const { pathname: path, hash, search } = url.parse(relativeUrl);
  return getAbsoluteUrl({ basePath: job.basePath, path, hash, search });
```

Reply:
I refactored this function from an existing compatabilityShim function and it had the happy path first so I just followed what it had. Since I already have a check for this before calling the function I am going to remove the check in the function.  So is the consensus that the error path should be first followed by the happy-path? Just so in the future I can be consistent